integrations/oci/sdk/runtime/src/main/java/io/helidon/integrations/oci/sdk/runtime/OciExtension.java

```diff
@@ -109,44 +114,124 @@
+     * necessarily be available. The configured authentication provider merely returns what is configured via
+     * {@value OciAuthenticationDetailsProvider#KEY_AUTH_STRATEGY} and/or
+     * {@value OciAuthenticationDetailsProvider#KEY_AUTH_STRATEGIES}. In order to additionally check if the provider is available,
+     * the {@code verifyIsAvailable} argument should be {@code true}.
+     *
+     * @param verifyIsAvailable flag to indicate whether the provider should be checked for availability
+     * @return the configured authentication type name
+     */
+    public static Class<? extends AbstractAuthenticationDetailsProvider>
+    configuredAuthenticationDetailsProvider(boolean verifyIsAvailable) {
+        return OciAuthenticationDetailsProvider.select(ociConfig(), verifyIsAvailable).authStrategy().type();
+    }
+
+    /**
+     * Returns the global {@link OciConfig} bean that is currently defined in the bootstrap environment.
+     * <p>
+     * The implementation will first look for an {@code oci.yaml} file, and if found will use that file to establish the global
+     * oci-specific bootstrap {@link io.helidon.config.spi.ConfigSource}.
+     * <p>
+     * If the implementation is unable to find this file, then a fallback mechanism will be used to find it in the configuration
+     * found in the {@link InjectionServices#globalBootstrap()}, using a top-level attribute key named
+     * {@value OciConfigBlueprint#CONFIG_KEY}.
+     * <p>
+     * The final fallback mechanism will use an {@code auto} authentication strategy - see {@link OciConfigBlueprint} for details.
      *
      * @return the bootstrap oci config bean
+     * @see OciConfigBlueprint
+     * @see #ociConfigSupplier
      */
     public static OciConfig ociConfig() {
-        Optional<Bootstrap> bootstrap = InjectionServices.globalBootstrap();
-        if (bootstrap.isEmpty()) {
-            LOGGER.log(System.Logger.Level.DEBUG, "No bootstrap - using default oci config");
-            return DEFAULT_OCI_CONFIG_BEAN.get();
-        }
+        io.helidon.common.config.Config config = configSupplier().get();
+        if (!config.exists()
+                || !(config.get(OciAuthenticationDetailsProvider.KEY_AUTH_STRATEGY).exists()
+                  || config.get(OciAuthenticationDetailsProvider.KEY_AUTH_STRATEGIES).exists())) {
+            // fallback
+            Optional<Bootstrap> bootstrap = InjectionServices.globalBootstrap();
+            if (bootstrap.isEmpty()) {
+                LOGGER.log(System.Logger.Level.DEBUG, "No bootstrap - using default oci config");
+                return DEFAULT_OCI_CONFIG_BEAN.get();
+            }
 
-        Config config = bootstrap.get().config().orElse(null);
-        if (config == null) {
-            LOGGER.log(System.Logger.Level.DEBUG, "No config in bootstrap - using default oci config");
-            return DEFAULT_OCI_CONFIG_BEAN.get();
-        }
+            config = bootstrap.get().config().orElse(null);
```

Comment:
This can be simplified to
```
config = InjectionServices.globalBootstrap()
        .flatMap(Bootstrap::config()
        .map(it -> it.get(OciConfig.CONFIG_KEY))
        .orElse(null);
if (config == null || !config.exists()) {
  ... return default
}
return OciConfig.create(config);
```
We do not need to know whether bootstrap is null, or config is not available on bootstrap - that is way too deep implementation detail on this component.

Reply:
I intended to keep the logging consistent to what Laird was doing in the MP implementation.

Reply:
pushed